lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -30,10 +43,62 @@
     BaseAPIController,
     UsesVisualizationMixin
 )
-from galaxy.work.context import WorkRequestContext
+from galaxy.work.context import (
+    SessionRequestContext,
+    WorkRequestContext,
+)
+from . import (
+    get_admin_user,
+    get_app,
+    get_trans,
+)
 
 log = logging.getLogger(__name__)
 
+router = APIRouter(tags=["jobs"])
+
+
+def get_job_manager(app: UniverseApplication = Depends(get_app)) -> JobManager:
+    return JobManager(app=app)
+
+
+def get_job_search(app: UniverseApplication = Depends(get_app)) -> JobSearch:
+    return JobSearch(app=app)
+
+
+def get_hda_manager(app: UniverseApplication = Depends(get_app)) -> hdas.HDAManager:
+    return app.hda_manager
+
+
+@cbv(router)
+class FastAPIJobs:
+    job_manager: JobManager = Depends(get_job_manager)
+    job_search: JobSearch = Depends(get_job_search)
+    hda_manager: hdas.HDAManager = Depends(get_hda_manager)
+
+    @router.get('/job_lock')
+    def job_lock_status(self, admin_user=Depends(get_admin_user)) -> JobLock:
```

Comment:
Is it intentional that this has a different route than the existing controller - because I think the existing route is api/job_lock not api/jobs/job_lock?

Reply:
No, just an oversight.